src/Modules/StatCan.OrchardCore.Hackathon/Services/HackathonService.cs

```diff
@@ -104,58 +114,65 @@ public async Task<ContentItem> JoinTeam(string teamContentItemId, ModelStateDict
                 return null;
             }
 
-            participant.Content.Hacker.Team = JObject.FromObject(new { ContentItemIds = new string[] { team.ContentItemId } });
+            var hacker = user.Properties.ToObject<ContentItem>();
+            hacker.Content.Hacker.Hacker.Team.ContentItemIds.Add(team.ContentItemId);
+            user.Properties = JObject.FromObject(hacker);
 
-            await _contentManager.UpdateAsync(participant);
+            _session.Save(user);
 
             return team;
         }
 
         public async Task<ContentItem> CreateTeam(ModelStateDictionary modelState)
         {
-            var participant = await GetParticipantFromSetAsync();
-            if (participant == null || participant.ContentType != "Hacker")
+            var user = await GetParticipantFromSetAsync();
+            if (user == null || !user.RoleNames.Contains("Hacker"))
             {
                 modelState.AddModelError("error", T["You are not a hacker"].Value);
                 return null;
             }
 
-            if (participant.HasTeam())
+            if (user.HasTeam())
             {
                 modelState.AddModelError("error", T["Already on a team"].Value);
                 return null;
             }
 
-            var user = _httpContextAccessor.HttpContext.User;
             var team = await _contentManager.NewAsync("Team");
-            team.Owner = user.Identity.Name;
+            team.Owner = user.UserId;
 
             await _contentManager.CreateAsync(team, VersionOptions.Published);
-
             await _contentManager.UpdateAsync(team);
-            participant.Content.Hacker.Team = JObject.FromObject(new { ContentItemIds = new string[] { team.ContentItemId } });
-            await _contentManager.UpdateAsync(participant);
+
+            var hacker = user.Properties.ToObject<ContentItem>();
+            hacker.Content.Hacker.Hacker.Team.ContentItemIds.Add(team.ContentItemId);
+            user.Properties = JObject.FromObject(hacker);
+
+            _session.Save(user);
 
             return team;
         }
 
         public async Task<bool> LeaveTeam(ModelStateDictionary modelState)
         {
-            var participant = await GetParticipantFromSetAsync();
-            if (participant == null || participant.ContentType != "Hacker")
+            var user = await GetParticipantFromSetAsync();
```

Comment:
Can you rename this method ot GetParticipanyAsync() the FromSet is an artifact of the old design.